src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,44 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(target) {
+    if (
+      !target
+      || !(target instanceof Herbivore)
+      || (target instanceof Carnivore)
```

Comment:
The condition to check if the target is a Carnivore is redundant because the previous condition already ensures that the target must be an instance of Herbivore. Remove the check for the Carnivore instance.